.expeditor/scripts/shared_release_habitat.sh

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .expeditor/scripts/shared.sh
+
+### This file should include things that are used exclusively by the release pipeline
+
+# Download public and private keys for the "core" origin from Builder.
+#
+# Currently relies on a global variable `hab_binary` being set, since
+# in the Linux build process, we need to switch binaries mid-way
+# through the pipeline. As we bring more platforms into play, this may
+# change. FYI.
+import_keys() {
+    echo "--- :key: Downloading 'core' public keys from ${HAB_BLDR_URL}"
+    ${hab_binary:?} origin key download core
+    echo "--- :closed_lock_with_key: Downloading latest 'core' secret key from ${HAB_BLDR_URL}"
+    ${hab_binary:?} origin key download \
+        --auth="${HAB_AUTH_TOKEN}" \
+        --secret \
+        core
+}
+
+get_latest_pkg_version_in_channel() {
+    local pkg_name="${1:?}"
+    version=$(curl -s "${HAB_BLDR_URL}/v1/depot/channels/core/$(get_release_channel)/pkgs/${pkg_name}/latest?target=${BUILD_PKG_TARGET}" \
+        | jq -r '.ident | .version')
+    echo "${version}"
+}
+
+# Install the latest binary from the release channel and set the `hab_binary` variable
+#
+# Requires the `hab_binary` global variable is already set!
+#
+# Accepts a pkg target argument if you need to override it, otherwise
+# will default to the value of `BUILD_PKG_TARGET`
+install_release_channel_hab_binary() {
+    local pkg_target="${1:-$BUILD_PKG_TARGET}"
+    curlbash_hab "${pkg_target}"
+
+    # TODO: workaround for https://github.com/habitat-sh/habitat/issues/6771	
+    ${hab_binary} pkg install core/hab-studio
+
+    echo "--- :habicat: Installed latest stable hab: $(${hab_binary} --version)"
+    # now install the latest hab available in our channel, if it and the studio exist yet
+
+    hab_version=$(get_latest_pkg_version_in_channel "hab")
+    studio_version=$(get_latest_pkg_version_in_channel "hab-studio")
+
+    if [[ -n $hab_version && -n $studio_version && $hab_version == "$studio_version" ]]; then
```

Comment:
Would `hab_version` or `studio_version` ever be set to the empty string? If not
```suggestion
    if [[ $hab_version == "$studio_version" ]]; then
```
should suffice.

Reply:
Yes there is a very real possibility that either or both could be empty string.

Reply:
I understand now. Still, 
```suggestion
    if [[ -n $hab_version && $hab_version == "$studio_version" ]]; then
```
should be enough, no?

Reply:
We could do `-n $studio_version && $hab_version == "$studio_version"` I suppose. I like being explicit about the fact that we want both to exist and match, and I think it makes the intent clearer.

Reply:
As a reader of the code, if a test seems logically redundant, I tend to think I must've misunderstood it. Reading this, I puzzled over whether there was a way for `-n $hab_version` and `$hab_version == "$studio_version"` to be true while `-n $studio_version` was false. If the goal is maximizing understandability, I tend to find minimizing the number of terms in a test to be generally helpful.

That said, it's not a biggie, so go with what you think it best; just wanted to share my experience as a reader.